tests/test_host.py

```diff
@@ -0,0 +1,104 @@
+"""Test host supervisor client."""
+
+from datetime import UTC, datetime
+
+from aioresponses import aioresponses
+from yarl import URL
+
+from aiohasupervisor import SupervisorClient
+from aiohasupervisor.models import HostOptions
+
+from . import load_fixture
+from .const import SUPERVISOR_URL
+
+
+async def test_host_info(
+    responses: aioresponses, supervisor_client: SupervisorClient
+) -> None:
+    """Test host info API."""
+    responses.get(
+        f"{SUPERVISOR_URL}/host/info", status=200, body=load_fixture("host_info.json")
+    )
+    result = await supervisor_client.host.info()
+    assert result.agent_version == "1.6.0"
+    assert result.chassis == "embedded"
+    assert result.virtualization == ""
+    assert result.disk_total == 27.9
+    assert result.disk_life_time == 10
+    assert result.features == [
+        "reboot",
+        "shutdown",
+        "services",
+        "network",
+        "hostname",
+        "timedate",
+        "os_agent",
+        "haos",
+        "resolved",
+        "journal",
+        "disk",
+        "mount",
+    ]
+    assert result.hostname == "homeassistant"
+    assert result.llmnr_hostname == "homeassistant3"
+    assert result.dt_utc == datetime(2024, 10, 3, 0, 0, 0, 0, UTC)
+    assert result.dt_synchronized is True
+    assert result.startup_time == 1.966311
+
+
+async def test_host_reboot(
+    responses: aioresponses, supervisor_client: SupervisorClient
+) -> None:
+    """Test host reboot API."""
+    responses.post(f"{SUPERVISOR_URL}/host/reboot", status=200)
+    assert await supervisor_client.host.reboot() is None
+    assert responses.requests.keys() == {("POST", URL(f"{SUPERVISOR_URL}/host/reboot"))}
+
+
+async def test_host_shutdown(
+    responses: aioresponses, supervisor_client: SupervisorClient
+) -> None:
+    """Test host shutdown API."""
+    responses.post(f"{SUPERVISOR_URL}/host/shutdown", status=200)
+    assert await supervisor_client.host.shutdown() is None
+    assert responses.requests.keys() == {
+        ("POST", URL(f"{SUPERVISOR_URL}/host/shutdown"))
+    }
+
+
+async def test_host_reload(
+    responses: aioresponses, supervisor_client: SupervisorClient
+) -> None:
+    """Test host reload API."""
+    responses.post(f"{SUPERVISOR_URL}/host/reload", status=200)
+    assert await supervisor_client.host.reload() is None
```

Comment:
Test options too?

Reply:
Think this is on the wrong one. `reload` has no options. But `reboot` does. And I also should create a separate model for its options in case we get other reboot options in the future that don't apply to shutdown like we did on the other one.